README.md

```diff
@@ -1,6 +1,161 @@
-# molecule-javascript
+# Molecule-JavaScript
 
-## Molecule
+## Quick Start
+
+There's a [template project](https://github.com/keith-cy/molecule-javascript-template) for quick start to use `molecule-javascript`.
+
+It shows the most general way of using `molecule-javascript` in Browsers and Node.js with real-world demos.
+
+## Start
+
+### Use in Node.js
+
+```js
+import Molecule from '@nervosnetwork/molecule-javascript'
+
+const normalizedSchema = {
+  name: 'Bytes',
+  type: 'fixvec',
+  item: {
+    type: 'byte',
+  },
+}
+const data = ['0x01', '0x02']
+const molecule = new Molecule(normalizedSchema)
+const serialized = molecule.serialize(data) // expect to be "0x020000000102"
+const parsed = molecule.deserialize('0x020000000102') // expect to be ["0x01", "0x02"]
+```
+
+### Use in Browsers
+
+```js
+const normalizedSchema = {
+  name: 'Bytes',
+  type: 'fixvec',
+  item: {
+    type: 'byte',
+  },
+}
+const data = ['0x01', '0x02']
+const molecule = new MoleculeJavaScript.Molecule(normalizedSchema)
+const serialized = molecule.serialize(data) // expect to be "0x020000000102"
+const parsed = molecule.deserialize('0x020000000102') // expect to be ["0x01", "0x02"]
+```
+
+### Normalize Schema
+
+Usually the schema will be defined with intermediate types in Molefile, e.g.
+
+```
+vector Word <byte>;
+vector Words <Word>;
+```
+
+There're two steps to normalize it.
+
+#### 1. Use [`Moleculec`](https://github.com/nervosnetwork/molecule/) to transform Molefile into JSON file, which is valid in JavaScript
+
+This step requires `rust`
+
+```bash
```

Comment:
Adding `git clone git@github.com:nervosnetwork/molecule.git` will be more better

Reply:
Why is `git clone` better than `cargo install`?

And this is consisitent with https://github.com/driftluo/moleculec-go/

Reply:
No. I mean add a shell comman not repalce the `cargo` command

Reply:
You mean there should be two ways to enable the `moleculec`?

Reply:
```
$ git clone git@github.com:nervosnetwork/molecule.git
$ cargo install moleculec
$ moleculec --language - --format json --schema-file ./types.mol > ./types.json
```

Reply:
> ```
> $ git clone git@github.com:nervosnetwork/molecule.git
> $ cargo install moleculec
> $ moleculec --language - --format json --schema-file ./types.mol > ./types.json
> ```

What does `git clone` do here?

Reply:
I've got your point. `moleculec` could be installed outside the origin repo.

Reply:
Where `moleculec ` is from ?

Reply:
> Where `moleculec ` is from ?

From `cargo`, it's the `npm` for `rust`

Reply:
OK.  I see. I just think `moleculec ` is from github repo in the past.